pkg/audit/verifier_test.go

```diff
@@ -523,3 +525,65 @@ func TestVerifierModifiedSegment(t *testing.T) {
 		assert.Empty(t, report)
 	})
 }
+
+func TestVerifierModifiedSegmentFailsOnce(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		err := planet.Satellites[0].Audit.Service.Close()
+		require.NoError(t, err)
+
+		ul := planet.Uplinks[0]
+		testData := testrand.Bytes(8 * memory.KiB)
+
+		err = ul.Upload(ctx, planet.Satellites[0], "testbucket", "test/path", testData)
+		require.NoError(t, err)
+
+		cursor := audit.NewCursor(planet.Satellites[0].Metainfo.Service)
+		stripe, _, err := cursor.NextStripe(ctx)
+		require.NoError(t, err)
+		require.NotNil(t, stripe)
+
+		verifier := audit.NewVerifier(zap.L(),
+			planet.Satellites[0].Metainfo.Service,
+			planet.Satellites[0].Transport,
+			planet.Satellites[0].Overlay.Service,
+			planet.Satellites[0].DB.Containment(),
+			planet.Satellites[0].Orders.Service,
+			planet.Satellites[0].Identity,
+			128*memory.B,
+			5*time.Second)
+
+		// delete the piece from the first node
+		origNumPieces := len(stripe.Segment.GetRemote().GetRemotePieces())
+		nodeID := stripe.Segment.GetRemote().GetRemotePieces()[0].NodeId
+		pieceID := stripe.Segment.GetRemote().RootPieceId.Derive(nodeID)
+		node := getStorageNode(planet, nodeID)
+		err = node.Storage2.Store.Delete(ctx, planet.Satellites[0].ID(), pieceID)
+		require.NoError(t, err)
+
+		report, err := verifier.Verify(ctx, stripe, nil)
+		require.NoError(t, err)
+
+		require.Len(t, report.Successes, origNumPieces-1)
+		require.Len(t, report.Fails, 1)
+		require.Equal(t, report.Fails[0], nodeID)
+		require.Len(t, report.Offlines, 0)
+		require.Len(t, report.PendingAudits, 0)
```

Comment:
These should be `assert` instead of `require`.

Reply:
Please justify this assertion.

Reply:
If this `require` fails it does not really prevent the rest of the test to execute, e.g. nothing will panic. Thus, we want it to be `assert` so the test case catches as many issues as possible. This often helps analyzing test failures.

Reply:
I really can't imagine wanting to fix more than one issue at once.  I feel like this is a personal preference, but I'll change it.

Reply:
Imagine a flaky test. When such a test fails, any bit of info is valuable. We may get such info from knowing how the rest of the asserts pass.